master/buildbot/changes/hgpoller.py

```diff
@@ -255,32 +248,37 @@ def _processChanges(self, unused_output):
         instead, we simply store the current rev number in a file.
         Recall that hg rev numbers are local and incremental.
         """
-        oid, current = yield self._getCurrentRev()
-        # hg log on a range of revisions is never empty
-        # also, if a numeric revision does not exist, a node may match.
-        # Therefore, we have to check explicitly that branch head > current.
-        head = yield self._getHead()
-        if head is None:
+        for branch in self.branches + self.bookmarks:
+            rev = yield self._getHead(branch)
+            if rev is None:
+                # Nothing pulled?
+                continue
+            yield self._processBranchChanges(rev, branch)
+
+    @defer.inlineCallbacks
+    def _processBranchChanges(self, newRev, branch):
+        prevRev = yield self._getCurrentRev(branch)
```

Comment:
nitpick: local variables and parameters should have `lower_case` capitalization.